lib/kubernetes-deploy/resource_watcher.rb

```diff
@@ -42,13 +42,19 @@ def report_what_just_happened(resources, watch_time)
       new_successes, new_failures = resources.partition(&:deploy_succeeded?)
       new_failures.each do |resource|
         if resource.deploy_failed?
+          status = "failure"
```

Comment:
Now that the classes are in charge of the reporting themselves, we could make this even simpler, i.e. right at the beginning of this method, `resources.each { |r| r.report_status_to_statsd(watch_time) }`. The resources themselves can easily figure out the status tag text.